packages/block-editor/src/components/heading-level-control/index.native.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { DropdownMenu } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import HeadingLevelIcon from './heading-level-icon';
+
+const HEADING_LEVELS = [ 1, 2, 3, 4, 5, 6 ];
+
+/** @typedef {import('@wordpress/element').WPComponent} WPComponent */
+
+/**
+ * HeadingLevelControl props.
+ *
+ * @typedef WPHeadingLevelControlProps
+ *
+ * @property {number} selectedLevel The chosen heading level.
+ * @property {(newValue:number)=>any} onChange Callback to run when toolbar value is changed.
+ * @property {boolean} isParagraphAllowed Append paragraph option with zero level to available levels.
+ */
+
+/**
+ * The `HeadingLevelControl` component renders a dropdown menu that displays
+ * heading level options for the selected block. This component is used to set
+ * the level of the heading in a block and also allows to use a `paragraph`
+ * element by setting the level to zero (`0`).
+ * This would also need handling in the block's `edit` function.
```

Comment:
To be clear, this line intends to state that a block's `edit` function would need to support a zero value, correct? If so, I found that train of thought to be difficult to follow. We might could add a bit of clarity. WDYT? 
```suggestion
 * element by setting the level to zero (`0`). The block's `edit` function would
 * also need to handle the zero value.
```